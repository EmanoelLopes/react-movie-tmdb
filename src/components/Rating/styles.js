import styled from 'styled-components';

export const VoteAverage = styled.span`
  background-color: #fff;
  display: inline-block;
  height: 25px;
  margin: 0 15px;
  max-width: 190px;
  position: relative;
  vertical-align: middle;
  width: 100%;

  small {
    background-color: #01d277;
    display: block;
    height: 100%;
  }

  small > span {
    margin-left: 5px;
  }
`;
