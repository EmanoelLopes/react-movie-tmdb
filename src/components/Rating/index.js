import React, { Fragment } from 'react';
import { number } from 'prop-types';
import { VoteAverage } from './styles';

const handleRatingStatyles = value => {
  return {
    width: `${value / 10 * 100}%`,
    backgroundColor: `${(value >= 7 && value < 10) ? '#01d277' : (value < 5) ? 'red' : 'orange' }`
  }
};

const Rating = ({ rating }) => {
  return (
    <Fragment>
      <VoteAverage>
        <small
          style={handleRatingStatyles(rating)}>
          <span>{rating}</span>
        </small>
      </VoteAverage> 
    </Fragment>
  );
};

Rating.propTypes = {
  rating: number.isRequired,
};

export default Rating;

