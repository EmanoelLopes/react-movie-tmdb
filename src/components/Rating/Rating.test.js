import React from 'react';
import { shallow } from 'enzyme';
import Rating from 'components/Rating';

let wrapped;
let defaultProps = {
  rating: 1,
};

beforeEach(() => {
  wrapped = shallow(<Rating {...defaultProps} />);
});

afterEach(() => {
  wrapped.unmount();
});

describe('<Rating />', () => {
  it('should match snapshot', () => {
    wrapped.setProps({ rating: 7 });
    expect(wrapped).toMatchSnapshot();
  });

  it('test backgroundColor green for average upper than 7', () => {
    wrapped.setProps({ rating: 7 });
    expect(wrapped.find('small')).toHaveStyle({ backgroundColor: '#01d277' });
  });

  it('test backgroundColor orange for average less than 7, greater than 5', () => {
    wrapped.setProps({ rating: 6 });
    expect(wrapped.find('small')).toHaveStyle({ backgroundColor: 'orange' });
  });

  it('test backgroundColor red for average less than 5', () => {
    wrapped.setProps({ rating: 4.9 });
    expect(wrapped.find('small')).toHaveStyle({ backgroundColor: 'red' });
  });
});
