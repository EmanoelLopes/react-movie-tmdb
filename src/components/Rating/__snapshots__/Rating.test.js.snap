// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> should match snapshot 1`] = `
<Fragment>
  <styled.span>
    <small
      style={
        Object {
          "backgroundColor": "#01d277",
          "width": "70%",
        }
      }
    >
      <span>
        7
      </span>
    </small>
  </styled.span>
</Fragment>
`;
